nativelib/src/main/scala/scala/scalanative/posix/sys/socket.scala

```diff
@@ -0,0 +1,162 @@
+package scala.scalanative.posix.sys
+
+import scala.scalanative.native.Nat._
+import scala.scalanative.native._
+import scala.scalanative.posix.sys.types.ssize_t
+import scala.scalanative.posix.sys.uio.iovec
+import scala.scalanative.runtime.time.timespec
+
+// http://man7.org/linux/man-pages/man7/socket.7.html
+
+/**
+ * Created by remi on 01/03/17.
+ */
+@extern
+object socket {
+
+  def socket(domain: CInt, _type: CInt, protocol: CInt): CInt = extern
+  def socketpair(domain: CInt,
+                 _type: CInt,
+                 protocol: CInt,
+                 sv: CArray[CInt, _2]): CInt = extern
+  def bind(sockfd: CInt, addr: Ptr[sockaddr], addrlen: socklen_t): CInt =
+    extern
+  def getsockname(sockfd: CInt,
+                  addr: Ptr[sockaddr],
+                  addrlen: Ptr[socklen_t]): CInt = extern
+  def connect(sockfd: CInt, addr: Ptr[sockaddr], addrlen: socklen_t): CInt =
+    extern
+  def getpeername(sockfd: CInt,
+                  addr: Ptr[sockaddr],
+                  addrlen: Ptr[socklen_t]): CInt = extern
+  def send(sockfd: CInt, buf: Ptr[Byte], len: CSize, flags: CInt): ssize_t =
+    extern
+  def sendto(sockfd: CInt,
+             buf: Ptr[Byte],
+             len: CSize,
+             flags: CInt,
```

Comment:
@densh Do we have a special construct for flags variables?

It would be useful to type this with value classes:

```
class SendToFlags(val underlying: Int) extends AnyVal {
  def &(other: SendToFlags) = new SendToFlags(underlying & other.underlying)
}

abstract class FlagValues[T] {
  private var current = 1
  def next() = {
    val t = current
    current = current << 1
    t
  }
}

object SendToFlags extends FlagValues[SendToFlags] {
  val MSG_CONFIRM = next()
  val MSG_DONTROUTE = next()
  val MSG_DONTWAIT = next()
  val MSG_EOR = next()
  val MSG_MORE = next()
  val MSG_NOSIGNAL = next()
  val MSG_OOB = next()
}

object Main extends App {
	import SendToFlags._

	println(MSG_CONFIRM)
	println(MSG_DONTROUTE)
	println(MSG_DONTWAIT)
	println(MSG_EOR)
	println(MSG_MORE)
	println(MSG_NOSIGNAL)
	println(MSG_OOB)
}

def sendto(sockfd: CInt, buf: Ptr[Byte], len: CSize, flags: SendToFlag, dest_addr: Ptr[sockaddr], addrlen: socklen_t): ssize_t = extern

```